File: binary_search.js
function binarySearch(arr,value){
    let start = 0;
    let end = arr.length -1;
    return searcher(arr,start,end,value)
}

function searcher(arr,start,end,value){
    let mid = Math.floor((start+end)/2);

    if(end <= start){
        return -1;
    }
    if(arr[mid] ===  value){
        return mid;
    }
    
    arr[mid] > value ? end = mid : (start = mid + 1, end = end + 1)
    return searcher(arr, start, end,value);
}

console.log(binarySearch([1,2,3,4,5,6,7,8,9,1,23,45,67,89,123,452,1667], 9))

File: PYTHON/binary_search.py
import math


def binary_search (arr,value) :
    start = 0
    end = len(arr) - 1
    while (end >= start) :
        mid = math.floor((start + end ) / 2)
        if(arr[mid] == value ) :
            return mid
        elif arr[mid] > value :
            end = mid - 1
        else :
            start = mid + 1

    return -1

print(binary_search([2,3,4,5,6,7,8,55,67], 13))

